src/test/java/activitySystem/ActivityTest.java

```diff
@@ -0,0 +1,43 @@
+package activitySystem;
+
+import com.gard.testbed.activitySystem.ActivityBuilder;
+
+import com.gard.testbed.activitySystem.IActivity;
+import com.gard.testbed.activitySystem.IBlackboard;
+import com.gard.testbed.activitySystem.Subject;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by va922kg on 3/15/16.
+ */
+public class ActivityTest {
+
+    @Test
+    public void createAnActivity() {
+        // using builder to create an activity
+        IActivity act = ActivityBuilder.activity("Omelette").addState("1").
+                setInitialState("1").
+                addTransition("1","2",(b -> b.query("eggs") > 3)).
```

Comment:
Here you can see the use of Predicate (input) -> output. In this way you don't need to create a class for each activity